config/Config-devel.in

```diff
@@ -62,6 +62,68 @@ menuconfig DEVEL
 		  Compile all host host tools even if not needed. This is needed to prepare a
 		  universal precompiled host tools archive to use in another buildroot.
 
+	menuconfig OPTIMIZE_HOST_TOOLS
+		bool "Host Tools Compile Options" if DEVEL
+
+	if OPTIMIZE_HOST_TOOLS
+
+		config HOST_FLAGS_OPT
+			string "Host Tools optimization flags"
+			depends on OPTIMIZE_HOST_TOOLS
```

Comment:
drop depends on OPTIMIZE_HOST_TOOLS as we already have the if for everything.

Reply:
Clause "`depends on OPTIMIZE_HOST_TOOLS`" ensures that options from this submenu doesn't appear in config if `CONFIG_OPTIMIZE_HOST_TOOLS` is not set. Same for `CONFIG_HOST_TOOLS_STRIP` and `CONFIG_HOST_FLAGS_STRIP`.

Reply:
Nono I verified. The "if OPTIMIZE_HOST_TOOLS" is enough to make the option not appear in the final .config

Indeed from the kernel kconfig documentation is precisely said that either if CONFIG block or depends on CONFIG should be used as they do the same exact thing. (I just verified that to make sure)